tests/lapi/coroutine_test.lua

```diff
@@ -0,0 +1,177 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+2.6 – Coroutines
+https://www.lua.org/manual/5.3/manual.html#2.6
+
+Computation of stack limit when entering a coroutine is wrong,
+https://github.com/lua/lua/commit/e1d8770f12542d34a3e32b825c95b93f8a341ee1
+
+C-stack overflow with deep nesting of coroutine.close,
+https://www.lua.org/bugs.html#5.4.4-9
+
+C stack overflow (again),
+https://github.com/lua/lua/commit/34affe7a63fc5d842580a9f23616d057e17dfe27
+
+When a coroutine tries to resume a non-suspended coroutine,
+it can do some mess (and break C assertions) before detecting the error,
+https://www.lua.org/bugs.html#5.3.3-4
+
+debug.getlocal on a coroutine suspended in a hook can crash the interpreter,
+https://www.lua.org/bugs.html#5.3.0-2
+
+Suspended __le metamethod can give wrong result,
+https://www.lua.org/bugs.html#5.3.0-3
+
+Resuming the running coroutine makes it unyieldable,
+https://www.lua.org/bugs.html#5.2.2-8
+
+pcall may not restore previous error function when inside coroutines,
+https://www.lua.org/bugs.html#5.2.1-2
+
+Wrong handling of nCcalls in coroutines,
+https://www.lua.org/bugs.html#5.2.0-4
+
+coroutine.resume pushes element without ensuring stack size,
+https://www.lua.org/bugs.html#5.1.3-2
+
+Recursive coroutines may overflow C stack,
+https://www.lua.org/bugs.html#5.1.2-4
+
+Stand-alone interpreter shows incorrect error message when the
+"message" is a coroutine,
+https://www.lua.org/bugs.html#5.1.2-12
+
+Debug hooks may get wrong when mixed with coroutines,
+https://www.lua.org/bugs.html#5.1-7
+
+Values held in open upvalues of suspended threads may be
+incorrectly collected,
+https://www.lua.org/bugs.html#5.0.2-3
+
+Attempt to resume a running coroutine crashes Lua,
+https://www.lua.org/bugs.html#5.0-2
+
+debug.getlocal on a coroutine suspended in a hook can crash the interpreter,
+https://www.lua.org/bugs.html#5.3.0-2
+
+debug.sethook/gethook may overflow the thread's stack,
+https://www.lua.org/bugs.html#5.1.2-13
+
+Memory hoarding when creating Lua hooks for coroutines,
+https://www.lua.org/bugs.html#5.2.0-1
+
+Synopsis:
+
+coroutine.close(co)
+coroutine.create(f)
+coroutine.isyieldable([co])
+coroutine.resume(co [, val1, ...])
+coroutine.running()
+coroutine.status(co)
+coroutine.wrap(f)
+coroutine.yield(...)
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+local CORO_OBJECTS
+
+-- Possible coroutine statuses, described in Lua 5.1 Reference Manual,
+-- https://www.lua.org/manual/5.4/manual.html#6.2
+local CORO_STATUS = {
+    DEAD = "dead",
+    NORMAL = "normal",
+    RUNNING = "running",
+    SUSPENDED = "suspended",
+}
+
+local CORO_ACTION_NAME = {
```

Comment:
Please, add a comment that `coroutine.status()` is used inside the target function.

Reply:
```diff
--- a/tests/lapi/coroutine_test.lua
+++ b/tests/lapi/coroutine_test.lua
@@ -80,7 +80,8 @@ local test_lib = require("lib")
 local CORO_OBJECTS = {}
 
 -- The function `coroutine.isyieldable()` is checked together with
--- `coroutine.yield()`.
+-- `coroutine.yield()`, `coroutine.status()` is used inside the
+-- target function.
 local CORO_ACTION_NAME = {
     "create",
     "resume",
```